geopandas/geoseries.py

```diff
@@ -173,8 +222,15 @@ def copy(self, order='C'):
         copy : GeoSeries
         """
         # FIXME: this will likely be unnecessary in pandas >= 0.13
-        return GeoSeries(self.values.copy(order), index=self.index,
-                      name=self.name).__finalize__(self)
+        return GeoSeries(self._data,
+                         index=self.index, crs=self.crs,
+                         name=self.name)
```

Comment:
We could do this for now. Or maybe we could also do for now, when setting something, that it takes a copy of the full VectorizedGeomtry? (which might be easier to track?) 
That of course makes setting something very expensive, but we could document that it is a limitation for now.

I would need to think a bit more on usecases to have an idea on how important it is to be able to set part of the geometries to new values.